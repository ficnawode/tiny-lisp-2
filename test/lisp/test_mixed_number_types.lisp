; int + int -> int
; Expected: 15
(print-debug (+ 10 5))

; float + float -> float
; Expected: 11.000000
(print-debug (+ 10.5 0.5))

; int + float -> float
; Expected: 15.500000
(print-debug (+ 10 5.5))

; float + int -> float
; Expected: 15.500000
(print-debug (+ 5.5 10))

; (- 10.0 (+ 2 3.5)) -> (- 10.0 5.5) -> 4.5
; Expected: 4.500000
(print-debug (- 10.0 (+ 2 3.5)))

(let ((my-int 100) (my-float 0.5))
  ; Expected: 99.500000
  (print-debug (- my-int my-float))
)