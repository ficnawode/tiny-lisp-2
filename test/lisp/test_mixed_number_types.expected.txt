15
11.000000
15.500000
15.500000
4.500000
99.500000